pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
@derekwaynecarr SGTM. But we still need the housekeeper to cleanup the pod is the pod failed to be killed but has already been marked as Failed (or Evicted). :)

And just food for thoughts, offline discussed with @yujuhong at noon. In the future, we may want to make the worker become a "real" worker. All the decision maker components just send tasks to the corresponding worker to order it do specific work. Then we can let the other components make the "real" decision. However, this may need a lot more consideration and refactoring. :)